setup.go

```diff
@@ -176,15 +223,24 @@ func getLogger(name string) *zap.SugaredLogger {
 	log, ok := loggers[name]
 	if !ok {
 		levels[name] = zap.NewAtomicLevelAt(zapCfg.Level.Level())
-		cfg := zap.Config(zapCfg)
-		cfg.Level = levels[name]
-		newlog, err := cfg.Build()
+		zapCfg.Level = levels[name]
+		newlog, err := zapCfg.Build()
 		if err != nil {
 			panic(err)
 		}
-		log = newlog.Named(name).Sugar()
+		log = newlog.Named(name).Sugar().With(fields...)
 		loggers[name] = log
 	}
 
 	return log
 }
+
+// Cleanup is for testing purposes only.
+// Cleanup resets the package state.
+func Cleanup() {
```

Comment:
Why do we export this?

Reply:
Because I was having an issue getting the `Example*` tests to work without them being in `package log_test` but it seems that it is no longer required. I will remove.